src/utils/child_process.ts

```diff
@@ -37,4 +39,25 @@ export function execChildProcess(command: string): Promise<ExecChildProcessResul
         });
     });
     return r;
+}
+
+
+export async function execFile(command: string, args: string[], options: child_process.ExecFileOptions): Promise<ExecChildProcessResult> {
+    return new Promise<ExecChildProcessResult>((resolve, reject) => {
+        child_process.execFile(command, args, {
+            encoding: 'utf8',
+            ...options
+        }, (error, stdout, stderr) => {
+            if (!!error) {
+                reject(error);
+                return;
+            }
+
+            resolve({
```

Comment:
This is giving me a type error when compiling:

```
[ts]
Argument of type '{ stdout: Buffer; stderr: Buffer; }' is not assignable to parameter of type 'ExecChildProcessResult<string> | PromiseLike<ExecChildProcessResult<string>> | undefined'.
  Type '{ stdout: Buffer; stderr: Buffer; }' is not assignable to type 'PromiseLike<ExecChildProcessResult<string>>'.
    Property 'then' is missing in type '{ stdout: Buffer; stderr: Buffer; }'.
```

Reply:
`encoding: 'utf8'` should make it's type string.

Did you set `encoding:??` in options argument? Typescript is dependently typed and tsc seems like doing inference across function boundaries.

Also making`encoding: 'utf8'` go last should fix the issue.

Reply:
I'm just compiling a clean checkout, no changes or new functions. The options comes from `process.env`, I'm not sure how I would get `encoding` into that at compile time, perhaps it is platform-specific? The code should really compile across all platforms if that is the case.

Reply:
I guess typescript compiler is insisting that type cannot be determined (so defaulted to Buffer) because `process.env`  *may* have `encoding` field with other value. `tsc` is right in this case but I don't know why I didn't get any error. Moving `encoding: 'utf8'` to last will fix the issue.

Reply:
I'll send a pr soon.